core/src/main/java/io/grpc/internal/AbstractServerStream.java

```diff
@@ -158,6 +157,11 @@ public final void cancel(Status status) {
     abstractServerStreamSink().cancel(status);
   }
 
+  @Override
+  public final boolean isReady() {
```

Comment:
It makes the method `final`. It used to be final in `AbstractStream2` but for making isReady() return `false` on cancellation it is now overridable. But it shouldn't need to be overridden in each transport.

Eventually, this method should probably also handle cancellation, but I didn't want to expand the scope of the PR since it is already large enough.